GetStream/stream-chat-android
# Remove keystroke and stopTyping extensions

Closes #3146 
### 🎯 Goal

To remove ChatDomain completely and improve the public API of ChatClient we need to move these extension functions into LLC itself.

### 🛠 Implementation details

- `ChatClient::keystroke` and `ChatClient::stopTyping` extensions are removed.
- Those methods are now in the main `ChatClient` class.
- `TypingEventListener` is added to run the side-effects for typing events.
- ChannelController tests are migrated or removed. 

### 🧪 Testing
- Open sample app in two devices.
- Start typing in one device.
- You should be able to see `typing.start` or `typing.stop` events in logcat. (There is some issue with UI not being updated but that is out of the scope of this PR as this issue is already present in `develop`)

### ☑️Contributor Checklist

#### General

#### Code & documentation

### ☑️Reviewer Checklist
